src/tests/MockProvider.ts

```diff
@@ -0,0 +1,88 @@
+import { readFile as _readFile } from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+
+const readFile = promisify(_readFile);
+import type { APIProvider, APIResponse } from '@wharfkit/session';
+
+const __dirname = path.resolve(path.dirname(''));
+
+export interface MockInterface {
+    url: string;
+    method: 'POST' | 'GET';
+    request: {
+        contentType: string;
+        payload: any;
+    };
+    response: {
+        code: number;
+        body: string | any;
+        location: string | null;
+    };
+}
+export class MockProvider implements APIProvider {
+    async load(filename: string): Promise<MockInterface | undefined> {
+        const fullPath = __dirname + '/src/tests';
+        const filePath = path.join(fullPath, filename);
+        return await this.getMockData(filePath);
+    }
+
+    mapFilename(path: string) {
```

Comment:
I think it'd be helpful to mark the private methods as private here since we're overriding APIProvider. My assumption is only the "call" method is actually part of the APIProvider interface.